File: java/gigasecond/src/main/java/Gigasecond.java
import java.time.Duration;
import java.time.LocalDate;
import java.time.LocalDateTime;

public class Gigasecond {
    LocalDateTime moment;

    public Gigasecond(LocalDate moment) {
        this.moment = moment.atStartOfDay();
    }

    public Gigasecond(LocalDateTime moment) {
        this.moment = moment;
    }

    public LocalDateTime getDateTime() {
        return moment.plus(Duration.ofSeconds(1_000_000_000));
    }
}

File: kotlin/gigasecond/src/main/kotlin/Gigasecond.kt
import java.time.LocalDate
import java.time.LocalDateTime
import java.time.LocalTime

class Gigasecond(localDateTime: LocalDateTime) {

    private val secondsInAGigasecond = 1_000_000_000L

    constructor(localDate: LocalDate, localTime: LocalTime = LocalTime.MIN) : this(
        LocalDateTime.of(
            localDate.year,
            localDate.month,
            localDate.dayOfMonth,
            localTime.hour,
            localTime.minute,
            localTime.second,
            localTime.nano
        )
    )

    val date: LocalDateTime = localDateTime.plusSeconds(secondsInAGigasecond)
}
